### PR TITLE
add ListPeers method that returns universe's fetchers.

### PR DESCRIPTION
**Pull request by manu844**
Currently there is no way for apps using galaxycache to unit test peers when using AddPeer/RemovePeer methods for incremental changes. ListPeers method returns fetchers map keyed by peer ID can be used to verify the peers and its address by using a RemoteFetcher implementation that provides required information.

### CODE REVIEW

File: peers.go [hunk 1]
```diff
@@ -395,6 +395,13 @@ func (pp *PeerPicker) remove(ids ...string) error {
 	return eg.Wait()
 }
 
+func (pp *PeerPicker) listPeers() map[string]RemoteFetcher {
+	pp.mu.Lock()
+	defer pp.mu.Unlock()
+	fetchers := pp.fetchers
+	return fetchers
```
Comment (dfinkel) [inline peers.go:402]: We need to make a copy here.
Since maps are pointer-ish, mutations to the map wouldn't be synchronized.

File: peers.go [hunk 2]
```diff
@@ -395,6 +395,13 @@ func (pp *PeerPicker) remove(ids ...string) error {
 	return eg.Wait()
 }
 
+func (pp *PeerPicker) listPeers() map[string]RemoteFetcher {
+	pp.mu.Lock()
+	defer pp.mu.Unlock()
```
Comment (dfinkel) [inline peers.go:400]: Let's use `RLock()` and `RUnlock()` here. (there's no reason to make this exclusive)

File: galaxycache.go [hunk 3]
```diff
@@ -232,6 +232,11 @@ func (universe *Universe) RemovePeers(ids ...string) error {
 	return universe.peerPicker.remove(ids...)
 }
 
+// ListPeers returns Universe's remote fetchers map keyed by peer IDs
```
Comment (dfinkel) [inline galaxycache.go:235]: Let's reword this to say that this returns "a map of remote fetchers keyed by Peer ID"

Maybe in a second doccomment line, we should mention that this is only expected to be useful when testing.

**Review by dfinkel — COMMENTED** [review]
I have no idea why github actions didn't run :man_shrugging: 

It would be nice to have some test-coverage, but I don't think it needs a whole new test.
Perhaps add a check in the incremental-peer handling test that calls this method and verifies that the result is what we expect.

File: galaxycache.go (hunk 3 above)
Comment (manu844) [reply galaxycache.go]: done

File: peers_test.go [hunk 4]
```diff
@@ -244,6 +244,14 @@ func TestPeersIncremental(t *testing.T) {
 					fetcherURIs[f.(*TestFetcher).uri] = struct{}{}
 				}
 
+				allFetchers := u.peerPicker.listPeers()
```
Comment (dfinkel) [inline peers_test.go:247]: Let's get some extra code-coverage area by calling `u.ListPeers()` here instead of going through the unexported interfaces. (there are times when going through private interfaces in tests is the right option, but if it's equivalent to an exported one, it provides extra coverage at zero-cost)

File: peers.go [hunk 5]
```diff
@@ -395,6 +395,19 @@ func (pp *PeerPicker) remove(ids ...string) error {
 	return eg.Wait()
 }
 
+func (pp *PeerPicker) listPeers() map[string]RemoteFetcher {
+	pp.mu.RLock()
+	defer pp.mu.RUnlock()
+
+	// deep copy of fetchers map with size equal to length of pp.fetchers.
```
Comment (dfinkel) [inline peers.go:402]: Hmm, this isn't actually a deep copy. (we're only copying the fetcher's pointer/interface-value)
It's a simple copy of the map. (also, I don't think we need the comment to mention that it's setting the size, since that's a pretty trivial detail that should be clear from the `make` call.

**Review by dfinkel — APPROVED** [review]
looks good.
Just one trivial comment on a comment.

File: peers.go (hunk 5 above)
Comment (manu844) [reply peers.go]: fixed the comment.
